src/GraphQL/ObjectExtensions.cs

```diff
@@ -135,7 +136,16 @@ public static object ConvertValue(object value, Type fieldType)
                 return value;
             }
 
-            var text = value.ToString();
+            string text;
```

Comment:
The point of this is to potentially convert the value to a specific type.  Perhaps there should be a more generic check if the `value` is already of the target `fieldType`?  Similar to what the `DateTime` check is doing above.  That would remove the need of the additional parse.  Of course if it does need the conversion then the below would be required.

Reply:
Yes, I agree. Not knowing the internals I went for the safest solution. For example, I did not know if the ObjectExtensions' method is used to convert between float and decimal. Or between float and strings, etc.
Depending on the requirements we could check the fileldType and use .NET's Convert which already handles float, double and decimal conversions.

Reply:
I think the whole parsing, coercing etc. needs an overhaul. Currently "converting" values is happening in way too many places, but I think that kind of refactoring is outside the scope of this PR.

Reply:
> convert between float and decimal

That is possible.  It all depends on how the data is de-serialized and what Type someone added to their property on their object.

> I think the whole parsing, coercing etc. needs an overhaul.

I agree with this - the primary purpose of some of this was to convert the JSON.NET types to actual types.  So the core framework didn't have a dependency on JSON.NET.  Otherwise you would have to be checking for JSON.NET types (JObject, JArray, etc.) throughout the codebase.

Reply:
So do we change this part somehow?